### PR TITLE
Spit-and-polish for Elliot's chameneos

### PR DESCRIPTION
**Pull request by bradcray**
Here, I made proposed changes for a next version of chameneos-redux to my study version, in approximate order of importance:

I wouldn't expect this to result in any behavioral or performance changes relative to Elliot's version -- it's mostly just code cleanup and organization.

* added Elliot's name to the byline due to his work in tuning it
* made spin limit into a config
* combined two print routines into one now that they're called consecutively
* removed 'Population' record which I think is arguably less useful now that main() has to do more juggling of the populations manually.  Moved contents either inline or into standalone procs
* hoisted colors10 to the top as a result and/or because it seems more "global"
* changed from underscores to camelCase
* removed some curly brackets from a simple if-then-else

I also updated Elliot's version (with his permission) to include all of these changes other than the removal of Population so that we could weigh the stylistic impacts on main() vs. having the record.  As part of this, we ended up simplifying the interface to Population to take a color array, removed the population size configs (because they're not really all that flexible anyway), and introducing a 'colors' field to the constructor in order to be able to set up the population properly.  (We tried adding an initializer to avoid having to store the colors field, but this ran afoul of issue #5935)

### CODE REVIEW

**Comment by bradcray** [comment]
@ronawho: See what you think about this version from a stylistic perspective.  You'll probably want to compare against your version rather than the previous version of mine (which is what GitHub will show).

@lydia-duncan will probably want to take a look as well due to her history with this benchmark.

File: test/studies/shootout/chameneos-redux/bradc/chameneosredux-blc.chpl [hunk 1]
```diff
@@ -36,82 +53,35 @@ proc printColorEquations() {
...
-  // place, and then creating per-chameneos tasks to have meetings.
-  //
-  proc holdMeetings(numMeetings) {
-    const place = new MeetingPlace(numMeetings);
-
-    coforall c in chameneos do           // create a task per chameneos
-      c.haveMeetings(place, chameneos);
-
-    delete place;
+proc print(chameneos) {
```
Comment (ronawho) [inline test/studies/shootout/chameneos-redux/bradc/chameneosredux-blc.chpl]: arg name `chameneos` confused me. What do you think about `chameneosArr` or `chameneosGroup` or something?

**Comment by ronawho** [comment]
👍 

In person I mentioned some nits and a slight preference for keeping the `Population` wrapper, but I don't feel strongly about any of that. i.e. I'd have no real objections to submitting this version

File: test/studies/shootout/chameneos-redux/bradc/chameneosredux-blc.chpl (hunk 1 above)
Comment (bradcray) [inline test/studies/shootout/chameneos-redux/bradc/chameneosredux-blc.chpl]: I updated this arg name to 'population' in 223fb2a

**Comment by lydia-duncan** [comment]
:+1:

**Pull request merged**